apps/app_voicemail.c

```diff
@@ -13394,6 +13547,171 @@ static int manager_list_voicemail_users(struct mansession *s, const struct messa
 	return RESULT_SUCCESS;
 }
 
+static int manager_get_mailbox_summary(struct mansession *s, const struct message *m)
+{
+	struct ast_vm_user *vmu = NULL;
+	const char *id = astman_get_header(m, "ActionID");
+	char actionid[128];
+	struct ast_vm_user svm;
+	int nummessages;
+
+	const char *context = astman_get_header(m, "Context");
+	const char *mailbox = astman_get_header(m, "Mailbox");
+
+	if ((ast_strlen_zero(context) || ast_strlen_zero(mailbox))) {
+		astman_send_error(s, m, "Need 'Context' and 'Mailbox' parameters.");
+		return 0;
+	}
+
+	actionid[0] = '\0';
+	if (!ast_strlen_zero(id)) {
+		snprintf(actionid, sizeof(actionid), "ActionID: %s\r\n", id);
+	}
+
+	/* find user */
+	memset(&svm, 0, sizeof(svm));
+	vmu = find_user(&svm, context, mailbox);
+	if (!vmu) {
+		/* could not find it */
+		astman_send_ack(s, m, "There is no voicemail user of the given info.");
+		return 0;
+	}
+
+	astman_send_listack(s, m, "Voicemail box detail will follow", "start");
+
+	/* walk through user's mailbox and append info for each message */
+	nummessages = append_vmbox_info_astman(s, vmu, "VoicemailBoxDetail", actionid);
+	if(nummessages < 0) {
+		ast_log(LOG_ERROR, "Could not append voicemail box info.");
```

Comment:
This doesn't contain enough information to be useful, such as the mailbox and context